### PR TITLE
Fix incorrect format of doubles in created IFC files

### PR DESCRIPTION
**Pull request by KonstantinLeontev**
[BUG]BIMx writes real numbers in files like this: 123,45.E-10 because of local user overrides. IFC viewers can't parse this, so we should get 123.45E-10 with dot instead of comma.
[ISSUE]"We recommend use "G" instead of the "R" format specifier, since in some cases "R" fails to successfully round-trip double-precision floating point values." - from here [https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings?view=netframework-4.8#the-general-g-format-specifier](url)
[CHANGED]Get rid of redundant . before E in real numbers with scientific notation for new IFC files.
[CHANGED]Format specifier changed to G.
[CHANGED]Constructor for CultureInfo object now set useUserOverride to false.
[ADDED]Test case for converting doubles to string.

### CODE REVIEW

File: Xbim.Common/Step21/Part21Formatter.cs [hunk 1]
```diff
@@ -32,28 +32,29 @@ public string Format(string fmt, object arg, IFormatProvider formatProvider)
...
+                //var rDoubleStr = dArg.ToString("R", StepText.DoubleCulture);
+                //var fixedDbl = double.Parse(rDoubleStr, StepText.DoubleCulture);
+                //var result = fixedDbl.ToString("R", StepText.DoubleCulture);
 
                 //decimal decArg = new Decimal(dArg);                                
                 // string result = decArg.ToString().ToUpper();
                 // string result = string.Format("{0:e22}", arg);
                 //string result = dArg.ToString("R", CultureInfo.CreateSpecificCulture("en-US"));
-                if (result.Contains(".")) return result;
+                //if (result.Contains(".")) return result;
```
Comment (martin1cerny) [inline Xbim.Common/Step21/Part21Formatter.cs]: It is not possible to comment out this and following lines. Many numbers would than be wrong.

Comment (KonstantinLeontev) [reply Xbim.Common/Step21/Part21Formatter.cs]: Yes, now I see that we need those lines to get integer number with dot in the end. Just uncommented this.

**Comment by KonstantinLeontev** [comment]
Thanks for review!
Now after some testing I see that proposed G17 formatting specifier works fine but resulting numbers may be don't really need such precision. For example <5.3E-06> -> <5.3000000000000001E-06> probably not so beauty for human reading.
About comma issue - you can see it, if you try to execute HelloWall example with doubles that have non-zero decimal parts on system that use "en-US" with comma for decimal divisor. Numbers will combine dots and commas. It seems that always better set useUserOverride to false for CultureInfo objects.

**Comment by martin1cerny** [comment]
Hi @KonstantinLeontev , thank you very much for the contribution. I'm sorry I only merged it now. I'm clearing the PRs before we implement IFC4.3 and realised this one hasn't been merged. Thank you again. I hope you are still using the Toolkit for your projects.

**Pull request merged**